src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -2016,21 +2058,32 @@ namespace smt::noodler {
 
         axiomatized_persist_terms.insert(e);
         STRACE("str", tout  << "handle contains " << mk_pp(e, m) << std::endl;);
+        ast_manager &m = get_manager();
+        expr *x = nullptr, *y = nullptr;
+        VERIFY(m_util_s.str.is_contains(e, x, y));
 
         // if contains is of the form (str.contains (str.substr value2 0 (+ n (str.indexof value2 "A" 0))) "A"), derive simpler constraints
         expr * ind = nullptr;
+        zstring str;
         if(expr_cases::is_contains_index(e, ind, m, m_util_s, m_util_a)) {
             expr_ref ind_eq(m.mk_eq( ind, m_util_a.mk_int(-1) ), m);
             expr_ref ind_leq(m_util_a.mk_le( ind, m_util_a.mk_int(-1) ), m);
             literal not_e = mk_literal(mk_not({e, m}));
             add_axiom({~mk_eq(ind, m_util_a.mk_int(-1), false), ~mk_literal(e) });
             add_axiom({mk_eq(ind, m_util_a.mk_int(-1), false), mk_literal(e) });
             return;
+        // if constains is of the form (str.constains str (str.at )) rewrite to a regular constaint
```

Comment:
```suggestion
        // if constains is of the form (str.constains strX (str.at ...)) rewrite to a regular constaint ((str.at ...) \in union of chars of strX)
```

This is what is happening here, right?

Reply:
Yes, fixed